packages/nvidia_nat_weave/src/nat/plugins/weave/fastapi_plugin_worker.py

```
@@ -91,24 +101,40 @@ async def _add_weave_feedback_route(self, app: FastAPI, builder: WorkflowBuilder
+                """Add reaction and/or comment feedback for an assistant message via observability trace ID."""
 
                 async with session_manager.session(http_connection=request,
                                                    user_authentication_callback=self._http_flow_handler.authenticate):
                     observability_trace_id = payload.observability_trace_id
                     reaction_type = payload.reaction_type
+                    comment = payload.comment
+
+                    # Validate that at least one feedback type is provided
+                    if not reaction_type and not comment:
+                        raise HTTPException(status_code=400,
+                                            detail="At least one of 'reaction_type' or 'comment' must be provided")
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)